src/python/pants/python/python_setup.py

```diff
@@ -183,19 +182,30 @@ def interpreter_search_paths(self):
     # changes
     @memoized_property
     def resolver_version(self) -> ResolverVersion:
-        if self.options.is_default("resolver_version"):
-            warn_or_error(
-                removal_version="2.4.0.dev0",
-                deprecated_entity_description="defaulting to the legacy pip resolver",
-                hint=(
-                    "In Pants 2.4.0, Pants will default to using pip's new resolver. The legacy "
-                    "resolver will be removed in Pants 2.5.\n\nSet "
-                    "`[python-setup].resolver_version = 'pip-2020-resolver` to prepare for the "
-                    "default changing. Refer to "
-                    "https://pip.pypa.io/en/latest/user_guide/#changes-to-the-pip-dependency-resolver-in-20-2-2020"
-                    " for more information on the new resolver."
-                ),
-            )
+        if not self.options.is_default("resolver_version"):
```

Comment:
This is okay, but it's much better imo to use the official help mechanism for deprecations because it will eagerly warn on a simple `./pants --version`, whereas this setup only warns when the codepath is used.

Reply:
But if the codepath is unused ... then the warning is noise right?

Reply:
Nope, the warning is solely if you manually configured the option. It doesn't display if you use the default.

Reply:
Another reason to use the help system mechanism, it will show up as deprecated in `./pants help-advanced` and on the docsite.

Reply:
Done.

Were you aware the help system does not wrap deprecations like it does everything else? I can file an issue or fix.

Reply:
Yeah I realized last week, but didn't file a ticket. That'd be helpful if you have the time